sources/packages/backend/apps/queue-consumers/src/processors/assessment/_tests_/cancel-application-assessment.processor.e2e-spec.ts

```diff
@@ -230,5 +242,277 @@ describe(
+            studentAssessmentStatus: StudentAssessmentStatus.Submitted,
+          },
+        },
+      );
+
+      // Queued job.
+      const job = createMock<Job<CancelAssessmentQueueInDTO>>({
+        data: { assessmentId: currentApplicationToCancel.currentAssessment.id },
+      });
+
+      // Act
+      const result = await processor.cancelAssessment(job);
+
+      // Asserts
+      expect(result.summary).toContain(
+        "No impacts were detected on future applications.",
+      );
+    });
+
+    it("Should find an impacted application and create a reassessment when the impacted application original assessment calculation date is after the cancelled assessment date original assessment, including 'Overwritten' records.", async () => {
+      // Arrange
+
+      // Create the student to be shared across all these applications.
+      const student = await saveFakeStudent(db.dataSource);
+      // Current application to be cancelled in Overwritten status because was edited.
+      const currentApplicationToCancelOverwritten =
+        await saveFakeApplicationDisbursements(
+          db.dataSource,
+          { student },
+          {
+            offeringIntensity: OfferingIntensity.partTime,
+            applicationStatus: ApplicationStatus.Overwritten,
+            currentAssessmentInitialValues: {
+              assessmentWorkflowId: "some fake id",
+              assessmentDate: addDays(-1, new Date()),
+              studentAssessmentStatus: StudentAssessmentStatus.Completed,
+            },
+          },
+        );
+      // Current application to be cancelled.
+      const currentApplicationToCancel = await saveFakeApplicationDisbursements(
+        db.dataSource,
+        { student },
+        {
+          offeringIntensity: OfferingIntensity.partTime,
+          applicationStatus: ApplicationStatus.Cancelled,
+          currentAssessmentInitialValues: {
+            assessmentWorkflowId: "some fake id",
+            assessmentDate: addDays(1, new Date()),
+            studentAssessmentStatus: StudentAssessmentStatus.CancellationQueued,
+          },
+        },
+      );
+      // Force pastOverwrittenApplication and pastApplication to share the same application number.
+      currentApplicationToCancel.applicationNumber =
+        currentApplicationToCancelOverwritten.applicationNumber;
+      await db.application.save(currentApplicationToCancel);
+      // Application in the future of the currentApplicationToCancelOverwritten but before
+      // the last assessment date from currentApplicationToCancel.
+      const impactedApplication = await saveFakeApplicationDisbursements(
+        db.dataSource,
+        { student },
+        {
+          offeringIntensity: OfferingIntensity.partTime,
+          applicationStatus: ApplicationStatus.Assessment,
+          currentAssessmentInitialValues: {
+            assessmentWorkflowId: "some fake id",
+            assessmentData: { weeks: 5 } as Assessment,
```

Comment:
Does adding `assessmentData` cause any impact?

Reply:
It does not, I just removed it 😉